plugins/CoreHome/Columns/VisitLastActionTime.php

```diff
@@ -35,9 +37,40 @@ class VisitLastActionTime extends VisitDimension
     protected $segmentName = 'visitServerHour';
     protected $acceptValues = '0, 1, 2, 3, ..., 20, 21, 22, 23';
 
+    /**
+     * Converts the hour to the hour depending on the configured site's timezone.
+     * Only works correct if a date/period is present in the request. Otherwise the result may vary depending on the
+     * day of the year as it can't know about daylight savings for example. Also if the currently selected date range
+     * includes multiple days with daylight savings and some without, then it might not be 100% correct as we are
+     * only looking at the start of the time.
+     * @param int $hour
+     * @param int $idSite
+     * @return int
+     * @throws \Exception
+     */
+    public static function convertHourToHourInSiteTimezone($hour, $idSite)
+    {
+        $date = Date::now();
+        $dateString = Common::getRequestVar('date', '', 'string');
+        $periodString = Common::getRequestVar('period', '', 'string');
+        if (!empty($dateString) && !empty($periodString)) {
+            try {
+                $date = Period\Factory::build($periodString, $dateString)->getDateStart();
+            } catch (\Exception $e) {
+                // ignore any error eg if values are wrong...
+            }
+        }
```

Comment:
Is it needed to use the date/period if we're just getting the hour later? Could we just use a dummy date or today()?

Reply:
@diosmosis  We'll try to use the date in case of daylight savings to make sure it is being considered. If we just used today then it wouldn't always be correct (see method description)

Reply:
I see, that's unfortunate.